lib/main.js

```diff
@@ -121,6 +121,12 @@ function initSession(shiny, options, shinyServer) {
 
       factory(url, ctx, (err, conn) => {
         pc.resolve(err, conn);
+        conn.onclose = e => {
```

Comment:
@jonyoder If you take another stab at this and want to write a test that verifies `onclose` doesn't get overwritten, you can maybe model it after this test: https://github.com/rstudio/shiny-server-client/blob/master/test/reconnect.js#L94-L97